src/components/SignIn.jsx

```diff
@@ -1,78 +1,82 @@
 import React from 'react';
+import PropTypes from 'prop-types';
+import { withRouter } from 'react-router-dom';
+
+import { auth } from '../firebase/firebase';
+import Button from './interface/Button';
+import DialogInput from './interface/dialog/DialogInput';
 
 class SignInPage extends React.Component {
-  state = {
-    name: ''
+  constructor(props) {
+    super(props);
+
+    const {
+      user,
+      history
+    } = props;
+
+    this.state = {
+      email: '',
+      password: ''
+    };
+
+    if (user) {
+      history.push('/events');
+    }
   }
 
-  handleNameChange = (event) => {
+  handleChange = (event) => {
+    const { target: { name, value }} = event;
     this.setState({
-      name: event.target.value
+      [name]: value
     });
   }
 
-  // TODO: Use firebase auth in this component
-
-  // handleSubmit = (event) => {
-  //   const { name } = this.state;
-
-  //   const item = {
-  //     name
-  //   };
-
-  //   try {
-  //     this.db.push(item);
-  //   } catch (e) {
-  //     alert(e);
-  //   }
-  //   alert(`submit worked: ${name}`);
-
-  //   event.preventDefault();
-  // }
-
-  // handleDataRetrieval = () => {
-  //   alert('suck it');
-  //   const vals = [];
-  //   this.db.once('value', (snapshot) => {
-  //     snapshot.forEach((data) => {
-  //       const value = {
-  //         name: data.val().name
-  //       };
-  //       vals.push(value);
-  //     });
-  //   });
-  //   console.log(vals);
-  // }
+  handleSubmit = () => {
+    const { history } = this.props;
+    const { email, password } = this.state;
+    auth.onAuthStateChanged((user) => {
+      if (user) {
+        history.push('events');
+      }
+    });
+    auth.signInWithEmailAndPassword(email, password);
+    return false;
+  }
 
   render() {
-    const { name } = this.state;
+    const { email, password } = this.state;
     return (
-      <div className="App">
-        <form onSubmit={this.handleSubmit}>
-          <label>
-            Name:
-            <input type="text" value={name} onChange={this.handleNameChange} />
-          </label>
+      <div>
+        <form>
+          <DialogInput
+            value={email}
+            name="email"
+            label="Email"
```

Comment:
label for email capitalizes the first letter i.e. `Email` while the password label is not capitalized i.e. `password`, we should keep it consistent with how all other labels